src/unix/darwin.c

```diff
@@ -130,9 +131,19 @@ uint64_t uv_get_constrained_memory(void) {
   return 0;  /* Memory constraints are unknown. */
 }
 
-
 uint64_t uv_get_available_memory(void) {
-  return uv_get_free_memory();
+  vm_statistics64_data_t info;
+  mach_msg_type_number_t count = sizeof(info) / sizeof(integer_t);
+
+  if (host_statistics64(mach_host_self(), HOST_VM_INFO64,
+                      (host_info64_t)&info, &count) != KERN_SUCCESS) {
+    return UV_EINVAL;  /* FIXME(bnoordhuis) Translate error. */
+  }
+  // Based on https://github.com/exelban/stats/blob/6c991de101957065f579a550ac2ae358d733d0c0/Modules/RAM/readers.swift#L47-L57
+  uint64_t used = info.active_count + info.inactive_count + info.compressor_page_count +info.speculative_count + info.wire_count;
+  uint64_t not_used = info.purgeable_count + info.external_page_count;
+  uint64_t available = uv_get_total_memory() - (used - not_used) * sysconf(_SC_PAGESIZE);
```

Comment:
I don't think `speculative_count` should be in `used`; since the kernel counts it as part of `free`: https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/osfmk/kern/host.c#L815

I think there's an easier way to figure out "available" memory:

```
uint64_t available = (info.free_count + info.inactive_count) * sysconf(_SC_PAGESIZE);
```

You could arguably include `purgeable_count` into that.  This completely ignores the external page count (which I believe represents pages which have been swapped to disk), and I think that.... is probably the right thing to do?  If macOS pages things out to make room for us, that's fine, and we can continue to use more physical memory.

Reply:
I'm not sure if external is swapped pages, or pages that are on disk by default, but the os is caching on RAM.